core/leader/leader.go

```diff
@@ -50,10 +44,45 @@ func Run(db *sql.DB, addr string, lead func(context.Context)) {
 	}
 	log.Messagef(ctx, "Using leaderKey: %q", l.key)
 
-	update(ctx, l)
-	for range time.Tick(5 * time.Second) {
-		update(ctx, l)
+	var leadCtx context.Context
+	var cancel func()
+	for leader := range leadershipChanges(ctx, l) {
```

Comment:
Sending a value over this channel raises the possibility that it might not strictly alternate. Would it be better to track that in this loop? like

```
leader := false
for range leadershipChanges(ctx, l) {
    leader = !leader
    if leader ...
}
```

Not sure, what do you think?